record/record_test.go

```diff
@@ -312,3 +313,145 @@ func TestStaleWriter(t *testing.T) {
 		t.Fatalf("stale write #1: unexpected error: %v", err)
 	}
 }
+
+func TestBasicReadRecovery(t *testing.T) {
+	records := [][]byte{
+		[]byte(strings.Repeat("a", blockSize-headerSize)),
+		[]byte(strings.Repeat("b", blockSize-headerSize)),
+		[]byte(strings.Repeat("c", blockSize-headerSize)),
+	}
+
+	buf := new(bytes.Buffer)
+	w := NewWriter(buf)
+
+	for i := 0; i < len(records); i++ {
+		wRec, err := w.Next()
+		if err != nil {
+			t.Fatal(err)
+		}
+		if _, err = wRec.Write(records[i]); err != nil {
+			t.Fatal(err)
+		}
+	}
+	w.Close() // Finalize the last record.
+
+	rawBufSlice := buf.Bytes()
+
+	// Corrupt the checksum of the second record in our file.
+	rawBufSlice[blockSize+0] = 0xef
+	rawBufSlice[blockSize+1] = 0xbe
+	rawBufSlice[blockSize+2] = 0xad
+	rawBufSlice[blockSize+3] = 0xde
+
+	// The first record should be read/processed just fine.
+	underlyingReader := bytes.NewReader(rawBufSlice)
+	r := NewReader(underlyingReader)
+	r1, err := r.Next()
```

Comment:
s/r1/r0/ might be more consistent with the comparison with records[0]. Similarly below.